### PR TITLE
ResultTable: HeaderCell and Cell rendering tweaks

### PR DESCRIPTION
**Pull request by stellarhoof**
- Do not wrap `HeaderCell` children in a `span` by default because it makes overriding `HeaderCell` harder.
- Support `headerCellProps` and `cellProps` to accomodate the very common pattern of needing to only pass props to the header/cell components.

### CODE REVIEW

File: packages/react/src/exampleTypes/ResultTable/index.js
Comment (positiveVibes4all) [inline packages/react/src/exampleTypes/ResultTable/index.js]: Shouldn't this span be removed also here?

Comment (stellarhoof) [reply packages/react/src/exampleTypes/ResultTable/index.js]: Indeed. Thanks for catching this 👍

File: packages/react/src/exampleTypes/ResultTable/Header.js
Comment (conraddavisjr) [inline packages/react/src/exampleTypes/ResultTable/Header.js]: A request to consolidate the components under a single guard since we're using `{!disableSort &&` twice.

File: packages/react/src/exampleTypes/ResultTable/Header.js
Comment (conraddavisjr) [inline packages/react/src/exampleTypes/ResultTable/Header.js]: Request to simplify
```suggestion
            onClick={() => mutate({ sortField, sortDir: 'desc' })}
```

File: packages/react/src/exampleTypes/ResultTable/Header.js
Comment (stellarhoof) [reply packages/react/src/exampleTypes/ResultTable/Header.js]: Done

File: packages/react/src/exampleTypes/ResultTable/Header.js
Comment (stellarhoof) [reply packages/react/src/exampleTypes/ResultTable/Header.js]: Done

**Review by conraddavisjr — APPROVED** [review]
There are quite a few things that I'm not in favor of regarding the structure of these components... and I suspect you feel similarly. However, this is an improvement, and one that we should make iteratively, instead of trying to do a huge lift. Thanks for your initiative on this @stellarhoof!

I'd like to get one more review & approval on this from @positiveVibes4all before merging.

**Review by positiveVibes4all — APPROVED** [review]
Looks good to go, this does not work with the current spark branch. I tested this on the draft PR in spark linked to this issue and it looked fine.

**Pull request merged**
